frontend/src/components/webVncDesktopLanding/WebVncDesktopLanding.tsx

```diff
@@ -0,0 +1,134 @@
+  const [url, setUrl] = useState("");
+
+  const initialiseVncServiceState = useCallback(async () => {
+    try {
+      const serviceData = await vncServiceStatus();
+      if (serviceData.isRunning) {
+        const urlData = await getVncDesktopUrl();
+        setUrl(urlData.url);
+        setVncServiceState(urlData.url !== "" ? VncServiceState.Running: VncServiceState.Stopped);
+      } else {
+        setVncServiceState(VncServiceState.Stopped);
+      }
+    } catch (_) {
+      return setVncServiceState(VncServiceState.Error);
+    }
+  }, []);
+
+  const { content, buttonLabel, buttonDisabled, onButtonClick } =
+    useMemo(() => {
+      switch (vncServiceState) {
+        case VncServiceState.Running:
+        return {
+          buttonLabel: "Let's Go!",
+          onButtonClick: () => window.open(url),
+          content: (
+            <>
+              Access programs and resources on your pi-top as if you were actually working on it!
+            </>
+          ), };
+        case VncServiceState.Stopped:
+          return {
+            buttonLabel: "Let's Go!",
+            buttonDisabled: true,
+            content: (
+              <>
+                The VNC service is not enabled in your device. Make sure to enable it and try again.
+                <br />
+                If your device is a pi-top[4], you can do this by navigating to the Settings menu in you miniscreen.
+              </>
+            ),
+          };
+
+        case VncServiceState.Error:
+          return {
+            buttonLabel: "Let's Go!",
+            buttonDisabled: true,
+            content: (
+              <>
+                There was an error while fetching your device state. Please try again later.
+              </>
+            ),
+          };
+
+        case VncServiceState.Unknown:
+          return {
+            content: (
+              <>
+                <p>Getting VNC service status...</p>
+                <Spinner size={45} />
+              </>
+            ),
+            buttonLabel: "Let's Go!",
+            buttonDisabled: true,
+          };
+      }
+    }, [vncServiceState, url]);
+
+  // initialise state on mount
+  useEffect(() => {
+    // don't bother initialising if on web renderer
+    if (runningOnWebRenderer()) return;
+
+    initialiseVncServiceState();
+  }, [initialiseVncServiceState]);
+
+  return (
+    <Layout
+      banner={{
+        src: keyboardScreen,
+        alt: "remote desktop banner",
+      }}
+      prompt={
+        <>
+          Want to access your device <span className="green">desktop</span>?
+        </>
+      }
+      nextButton={{
+        label: buttonLabel,
+        disabled: buttonDisabled,
+        onClick: onButtonClick,
+      }}
+      className={styles.root}
+      showHeader={false}
+    >
+      {
+        <div className={styles.content}>
+          {runningOnWebRenderer() ? (
+            <p>
+              Remote desktop cannot be used when on your pi-top. Open{" "}
```

Comment:
why doesn't it work? not that it would be useful... they should just use the desktop if they're on it.

Reply:
this is what happens when you open the remote-desktop on the device:

![image](https://user-images.githubusercontent.com/14126805/184984675-1ce0d497-42bb-49e1-b1c7-850b85434f90.png)